decompiler/pipeline/preprocessing/missing_definitions.py

```diff
@@ -143,13 +144,13 @@ def insert_missing_definitions(self):
                 self._insert_definition_if_undefined(variable, previous_ssa_labels, undefined_variables)
                 previous_ssa_labels.add(variable.ssa_label)
 
-    def _get_undefined_variables(self) -> Set[Variable]:
+    def _get_undefined_variables(self) -> InsertionOrderedSet[Variable]:
         """
         Compute the set of undefined variables.
 
         -> We assume that every aliased variable has to be defined for every memory version.
         """
-        undefined_variables = self._use_map.used_variables - self._def_map.defined_variables
+        undefined_variables: InsertionOrderedSet[Variable] = self._use_map.used_variables - self._def_map.defined_variables
```

Comment:
I think we need couple more changes here, even though it does not impact missing definitions directly.

The original problem with non-deterministic behavior does not occur anymore, but there are some inconsistencies in code.
Namely, I tested a bit and found out, that `InsertionOrderedSet - InsertionOrderedSet` is still `InsertionOrderedSet`, but it does not hold for `|`  (line 154) which should not be the case.

So I think we need to update `InsertionOrderSet` class to ensure that the operations on `InsertionOrderedSet` return `InsertionOrderSet` (`union`, `difference`, etc.). And there are no tests for operations on `InsertionOrderSet` for the moment, so those should be added.